### PR TITLE
Bridge profile identifier to Objective-C

### PR DESCRIPTION
**Pull request by frederoni**
Global variables defined in Swift doesn't bridge to Objective-C.

### CODE REVIEW

File: MapboxDirections/MBRouteOptions.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#import <Foundation/Foundation.h>
+
+
+/**
+ The returned directions are appropriate for driving or riding a car, truck, or motorcycle.
+ 
+ This profile prioritizes fast routes by preferring high-speed roads like highways. A driving route may use a ferry where necessary.
+ */
+extern NSString *const MBDirectionsProfileIdentifierAutomobile;
```
Comment (1ec5) [inline MapboxDirections/MBRouteOptions.h]: While you’re at it, can you make these constants into an extensible string enumeration?

**Comment by frederoni** [comment]
Rebased onto the `swift3` branch.

**Comment by 1ec5** [comment]
Actually, for consistency, it'd be better to rebase atop master and merge into swift3, as we've done for all the other changes.

File: MapboxDirections/MBRouteOptions.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#import <Foundation/Foundation.h>
+
+#pragma mark - Specifying the Routing Profile
+
+typedef NSString * MBDirectionsProfileIdentifier NS_EXTENSIBLE_STRING_ENUM;
```
Comment (1ec5) [inline MapboxDirections/MBRouteOptions.h]: This symbol could use some documentation.

File: MapboxDirections/MBRouteOptions.swift [hunk 3]
```diff
@@ -144,8 +110,8 @@ open class RouteOptions: NSObject {
         assert(waypoints.count <= 25, "A route may not have more than 25 waypoints.")
         
         self.waypoints = waypoints
-        self.profileIdentifier = profileIdentifier ?? MBDirectionsProfileIdentifierAutomobile
-        self.allowsUTurnAtWaypoint = ![MBDirectionsProfileIdentifierAutomobile, MBDirectionsProfileIdentifierAutomobileAvoidingTraffic].contains(self.profileIdentifier)
+        self.profileIdentifier = profileIdentifier ?? MBDirectionsProfileIdentifier.automobile.rawValue
+        self.allowsUTurnAtWaypoint = ![MBDirectionsProfileIdentifier.automobile.rawValue, MBDirectionsProfileIdentifier.automobileAvoidingTraffic.rawValue].contains(self.profileIdentifier)
```
Comment (1ec5) [inline MapboxDirections/MBRouteOptions.swift]: Should profileIdentifier be of type MBDirectionsProfileIdentifier?

Comment (frederoni) [inline MapboxDirections/MBRouteOptions.swift]: It's a breaking change and it involves quite a few changes but it would make things clearer.

Comment (1ec5) [inline MapboxDirections/MBRouteOptions.swift]: This library hasn't reached 1.0 yet, so breaking changes are perfectly fine.

**Comment by frederoni** [comment]
Temporarily enabled carthage build step solved the bitrise issues for tvOS, macOS & iOS.
OHHTTPStubs and Polyline doesn't have a scheme for watchOS so we have to upstream a fix or figure out another solution.

**Comment by 1ec5** [comment]
> OHHTTPStubs and Polyline don't have a watchOS target

Polyline does have a watchOS target, but it looks like a watchOS scheme was omitted – probably an oversight.

**Pull request merged**